### DISCUSSION TITLE
CSLA 7.0.3 ApplicationContext.Principal [ClassInitialize] vs [TestInitialize]

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#4122] [Questions]
Previously, I was setting `ApplicationContext.Principal` for our tests in `[ClassInitialize]` and this worked fine. At some point, all of the tests stopped working as the `Principal` had disappeared by the time a test method started resulting in the `IsInRole` auth failing. I've switched to setting up with `[TestInitialize]` which has resolved the issue, but I'm just wondering what changed?

**Comment by rockfordlhotka** [comment]
In the CSLA unit tests we tend to use `TestInitialize` so each test has its own DI setup and scope to provide isolation from the other tests.
